### PR TITLE
[cont_link_flap] address issue in fixture bring_up_fanout_interfaces

### PR DESCRIPTION
**Pull request by yxieca**
### Description of PR

Summary:

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### What is the motivation for this PR?
test_cont_link_flap was executed N times, where N is number of ports of the testbed.

#### How did you do it?
Fixture mistaenly uses all_ports enumerator, which causes test_cont_link_flap itself being parameterized with all available ports on DUT. So the test was executed N times instead of once. Where N is number of ports.

Also the warning in candidate builder is intended to give warning when port enumeration failed. Add another explicit value to get all ports without triggering the warning.

Signed-off-by: Ying Xie <ying.xie@microsoft.com>

#### How did you verify/test it?
Run link flap tests.

platform linux2 -- Python 2.7.12, pytest-4.6.5, py-1.9.0, pluggy-0.13.1
ansible: 2.8.12
rootdir: /var/src/sonic-mgmt/tests, inifile: pytest.ini
plugins: ansible-2.2.2, forked-1.3.0, xdist-1.28.0, html-1.22.1, repeat-0.8.0, metadata-1.10.0
collected 69 items                                                                                                                                                                       

platform_tests/link_flap/test_cont_link_flap.py::TestContLinkFlap::test_cont_link_flap Loading callback plugin json of type stdout, v2.0 from /usr/local/lib/python2.7/dist-packages/ansible/plugins/callback/json.pyc
Loading callback plugin json of type stdout, v2.0 from /usr/local/lib/python2.7/dist-packages/ansible/plugins/callback/json.pyc
FAILED                                                                                      [  1%]Loading callback plugin json of type stdout, v2.0 from /usr/local/lib/python2.7/dist-packages/ansible/plugins/callback/json.pyc
Loading callback plugin json of type stdout, v2.0 from /usr/local/lib/python2.7/dist-packages/ansible/plugins/callback/json.pyc

platform_tests/link_flap/test_cont_link_flap.py::TestContLinkFlap::test_cont_link_flap ERROR                                                                                       [  1%]
platform_tests/link_flap/test_link_flap.py::test_link_flap[str2-7050cx3-acs-06|Ethernet8] PASSED                                                                                   [  2%]
platform_tests/link_flap/test_link_flap.py::test_link_flap[str2-7050cx3-acs-06|Ethernet0] SKIPPED                                                                                  [  4%]
platform_tests/link_flap/test_link_flap.py::test_link_flap[str2-7050cx3-acs-06|Ethernet4] PASSED                                                                                   [  5%]
platform_tests/link_flap/test_link_flap.py::test_link_flap[str2-7050cx3-acs-06|Ethernet108] SKIPPED                                                                                [  7%]
platform_tests/link_flap/test_link_flap.py::test_link_flap[str2-7050cx3-acs-06|Ethernet100] SKIPPED                                                                                [  8%]
platform_tests/link_flap/test_link_flap.py::test_link_flap[str2-7050cx3-acs-06|Ethernet104] SKIPPED                                                                                [ 10%]
platform_tests/link_flap/test_link_flap.py::test_link_flap[str2-7050cx3-acs-06|Ethernet68] PASSED                                                                                  [ 11%]
platform_tests/link_flap/test_link_flap.py::test_link_flap[str2-7050cx3-acs-06|Ethernet96] PASSED                                                                                  [ 13%]
platform_tests/link_flap/test_link_flap.py::test_link_flap[str2-7050cx3-acs-06|Ethernet124] PASSED                                                                                 [ 14%]
platform_tests/link_flap/test_link_flap.py::test_link_flap[str2-7050cx3-acs-06|Ethernet92] PASSED                                                                                  [ 15%]
platform_tests/link_flap/test_link_flap.py::test_link_flap[str2-7050cx3-acs-06|Ethernet120] PASSED                                                                                 [ 17%]
platform_tests/link_flap/test_link_flap.py::test_link_flap[str2-7050cx3-acs-06|Ethernet56] PASSED

### CODE REVIEW

File: tests/platform_tests/link_flap/conftest.py [hunk 1]
```diff
@@ -26,23 +25,20 @@ def pytest_addoption(parser):
...
-        dutname, portname = decode_dut_port_name(all_ports)
-
         for dut in duthosts:
-            if dutname == 'unknown' or dutname == dut.hostname:
-                candidates = build_test_candidates(dut, fanouthosts, portname)
-                for _, fanout, fanout_port in candidates:
-                    fanout.no_shutdown(fanout_port)
+            candidates = build_test_candidates(dut, fanouthosts, "all_ports")
+            for _, fanout, fanout_port in candidates:
+                fanout.no_shutdown(fanout_port)
```
Comment (daall) [inline tests/platform_tests/link_flap/conftest.py:42]: I wonder if, just to be safe, we shouldn't condition this on `if request...failed`? Would it hurt to just bring up all the fanout ports at the end every time? (this doesn't need to be changed in this PR)

Comment (daall) [reply tests/platform_tests/link_flap/conftest.py:42]: Also, can we swap that `sleep` for a `wait_until` where we check if all the active DUT ports are showing as oper up? (this can also be improved upon later)

File: tests/platform_tests/link_flap/test_cont_link_flap.py [hunk 2]
```diff
@@ -73,7 +73,7 @@ def test_cont_link_flap(self, request, duthost, fanouthosts, bring_up_fanout_int
         # Flap all interfaces one by one on Peer Device
         for iteration in range(3):
             logging.info("%d Iteration flap all interfaces one by one on Peer Device", iteration + 1)
-            candidates = build_test_candidates(duthost, fanouthosts, 'unknown')
+            candidates = build_test_candidates(duthost, fanouthosts, 'all_ports')
```
Comment (daall) [inline tests/platform_tests/link_flap/test_cont_link_flap.py:76]: I think I'm missing something - why is this changed from unkown to all_ports?

**Review by daall — APPROVED** [review]
Some nits + 1 quick question, feel free to address in a follow-up PR.

File: tests/platform_tests/link_flap/conftest.py (hunk 1 above)
Comment (yxieca) [reply tests/platform_tests/link_flap/conftest.py:42]: Yeah. This is a stop bleeding change. There is more room for improvement.

File: tests/platform_tests/link_flap/test_cont_link_flap.py (hunk 2 above)
Comment (yxieca) [reply tests/platform_tests/link_flap/test_cont_link_flap.py:76]: To mute the warning message in candidate list build.
